cow_py/app_data/utils.py

```diff
@@ -0,0 +1,34 @@
+from re import T
+from typing import Any, Dict
+import httpx
+from multiformats import CID
+
+import json
+
+from cow_py.app_data.consts import DEFAULT_IPFS_READ_URI
+
+# CID uses multibase to self-describe the encoding used (See https://github.com/multiformats/multibase)
+#   - Most reference implementations (multiformats/cid or Pinata, etc) use base58btc encoding
+#   - However, the backend uses base16 encoding (See https://github.com/cowprotocol/services/blob/main/crates/app-data-hash/src/lib.rs#L64)
+MULTIBASE_BASE16 = "f"
+
+
+def extract_digest(cid_str: str) -> str:
+    # TODO: Verify this
+    cid = CID.decode(cid_str)
+    if cid_str[0] == MULTIBASE_BASE16:
+        return "0x" + cid.digest.hex()[4:]
+
+    return "0x" + cid.set(base="base58btc").digest.hex()[4:]
+
+
+def stringify_deterministic(obj):
+    return json.dumps(obj, sort_keys=True, separators=(",", ":"))
```

Comment:
`#could` add tests cases from [here ](https://github.com/haochi/json-stable-stringify-python/blob/master/tests/test_stringify.py)

Reply:
Maybe this [ref](https://github.com/gnosis/prediction-market-agent-tooling/blob/0b0e29529fc51ba1a16ae5d5754bc0de3a15f247/prediction_market_agent_tooling/tools/web3_utils.py#L307) helps?